cmd/kube-proxy/app/server.go

```diff
@@ -399,6 +399,20 @@ func NewProxyServer(config *componentconfig.KubeProxyConfiguration, cleanupAndEx
 	proxyMode := getProxyMode(string(config.Mode), iptInterface, iptables.LinuxKernelCompatTester{})
 	if proxyMode == proxyModeIPTables {
 		glog.V(0).Info("Using iptables Proxier.")
+		var nodeIP net.IP
+		if config.BindAddress == "0.0.0.0" || config.BindAddress == "" {
```

Comment:
Would it make more sense to check ip.IsUnspecified(), instead of specific strings (as with v6 it could be written multiple ways), and do that as part of the validation of the bind address on L406+. IOW, if no address specified, use getNodeIP(), if cannot parse return error, and if is unspecified or not local, return error.